__init__.py

```diff
@@ -288,45 +288,47 @@ def watch_dir_forever(self):
         # in order to show changes in relative rarity for blocks not touched by new traces
         idle = -1  # idle -1 means no new coverage seen
         idle_threshold = 5
+        ctx = get_ctx(self.bv)
         while True:
             dir_files = os.listdir(self.coverage_dir)
             new_files = [name for name in dir_files if name not in self.files_processed]
             new_coverage = set()
             for new_file in new_files:
-                new_coverage |= covdb.add_file(os.path.join(self.coverage_dir, new_file))
+                new_coverage |= ctx.covdb.add_file(os.path.join(self.coverage_dir, new_file))
                 log.log_debug("[DBG] Added new coverage from file %s @ %d" % (new_file, int(time())))
                 self.files_processed.append(new_file)
             num_new_coverage = len(new_coverage)
             if num_new_coverage > 0:
-                highlight_set(new_coverage)
+                highlight_set(self.bv, new_coverage)
                 idle = 0
                 log.log_debug("[DBG] Updated highlights for %d blocks" % num_new_coverage)
             else:
                 if idle >= 0:
                     idle += 1
                 if idle > idle_threshold:
-                    highlight_set(covdb.total_coverage)
+                    highlight_set(self.bv, ctx.covdb.total_coverage)
                     idle = -1
             sleep(1)
-            if not watching:
+            if ctx.watcher:
```

Comment:
This seems odd now that "watching" is gone, because `ctx.watcher` will be this `BackgroundHighlighter` instance, right? Does `watch_forever` still work? I'd have to test, but I think this if condition was originally a hedge in case `ctx.watcher.finish` didn't set `self.cancelled` (I'm not sure if it does or doesn't).

Reply:
Oops ya this should be `if ctx.watcher is None`